PyInstaller/hooks/hook-itk.py

```diff
@@ -0,0 +1,7 @@
+from PyInstaller.utils.hooks import collect_data_files
+
+hiddenimports = ['new']
+
+# If ITK is pip installed, gets all the files.
+itk_datas = collect_data_files('itk', include_py_files=True)
+datas = [x for x in itk_datas if '__pycache__' not in x[0]]
```

Comment:
This looks very ugly. Please check PyInstaller.utils.hooks for a more appropriate collection function. What shall be collected here in the end?

Why is it required to *unconditionally* include *all* files in this name-space? Even if all these modules would be required *unconditionally*, this will add them as files and not as precompiled modules.

Reply:
Unfortunately, because ITK is primarily developed in C++, its interfaces and how to use it is not always very Pythonic. In this case, the import process is different from what is usually done in Python packages (lazy loading, monolithic package). Even though we, ITK developers, are trying to improve some of this, some decisions were made a long time ago and are not easy to address.

Because of the way ITK is wrapped (from C++) in Python, a set of non-conventional files are generated and required for ITK to work in Python. This includes:
* The dynamic libraries.
* Python modules that call functions in the dynamic libraries.
* Additional configuration files that describe the modules and how to load them.

Despite these limitations, including all the (required) files allows to use PyInstaller with ITK seamlessly, which is what some ITK users have ask us to help them with.

Reply:
Note that since ITK is divided into smaller subpackages to be able to be uploaded on PyPI, an advanced user that knows exactly what is required could install only the required ITK subpackages. However, due to the lazy-loading and monolithic packaging of ITK, there is no way to find what part of the package is required simply looking at the `import` statements.